transformations/use_acronyms/transformation.py

```diff
@@ -0,0 +1,94 @@
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+"""
+Use Acronyms.
+    Change groups of words for their equivalent acronym.
+    The acronyms were scraped from Wikipedia's List of Acronyms and naively filtered.
+"""
+
+
+def transformation(sentence, lowercase, acronyms):
+    new_sentence = sentence
+    lower_sentece = sentence.lower()
+    for key in acronyms.keys():
+        lower_key = key.lower()
+        if key in new_sentence:
+            new_sentence = new_sentence.replace(key, acronyms[key])
+        elif lowercase and (lower_key in lower_sentece):
+            key_index = lower_sentece.index(lower_key)
+            key_end = key_index + len(key)
+            new_sentence = (
+                new_sentence[:key_index]
+                + acronyms[key]
+                + new_sentence[key_end:]
+            )
+
+    return new_sentence
+
+
+class UseAcronyms(SentenceOperation):
+    tasks = [TaskType.TEXT_CLASSIFICATION, TaskType.TEXT_TO_TEXT_GENERATION]
+    languages = ["en"]
+    keywords = [
+        "lexical",
+        "rule-based",
+        "external-knowledge-based",
+        "tokenizer-required",
+        "high-precision",
+        "low-coverage",
+        "low-generations",
+    ]
+
+    def __init__(
+        self,
+        seed=0,
+        max_outputs=1,
+        lowercase=False,
+        acronyms_file="transformations/use_acronyms/acronyms.tsv",
```

Comment:
This could be improved to make it more robust and less dependent on your working directory, e.g. by using a file path instead.  

E.g.:
At the top `import os`
and your input `acronyms_file="acronyms.tsv"`
And then accordingly:
       ` acronyms_path=os.path.join(
                os.path.dirname(os.path.abspath(__file__)),
                acronyms_file)`

And then later:
`with open(acronyms_path, "r", encoding=encoding) as file:`

Reply:
I wanted to make it possible for the user to use their own list of acronyms (that way it can work on a very specific set f acronyms or with data in other languages). But I will apply this suggestion to make sure that when loading the default file it will be done properly.

Reply:
Yes, it sounds reasonable. I am happy for you to leave it as is.

Reply:
Perfect, now the default acronyms file is being loaded using the file path as suggested while still allowing for user defined pths for use defined files.

I also added a bit of code to make sure the dictionary keys are ordered from larger to shorter, to prevent smaller Acronyms to override larger ones (ie. prevent "New York University" from being turned into "NY University" instead of "NYU").